### DISCUSSION TITLE
/articles/better-node-js-install-management-with-node-version-manager/

### DISCUSSION THREAD

**Discussion by vtns-comment-importer** [Voitanos/voitanos.io-comments#162] [Comments]
Andrew Connell, MVP SharePoint Server - Better Node.js Install Management with Node Version Manager
https://www.andrewconnell.com/articles/better-node-js-install-management-with-node-version-manager/

**Comment by vtns-comment-importer** [comment]
```header
Originally posted by Bob on 2017-06-06 05:32
```

Great article Andrew, thanks for sharing.

Just a point about Administrative Command Prompts issue - every time I change the used version using "nvm use v8" I get a prompt for admin credentials. Is this because it wants to change the PATH to node to this version?

**Reply by vtns-comment-importer** [reply]
```header
Originally posted by Andrew Connell on 2017-06-06 06:30
```

I’m assuming you’re on Windows? If so, I can’t speak to that because I’m not as familiar with the Windows port of NVM… looks like it’s just as aspect of the Windows port from this issue: [https://github.com/coreybut...](https://github.com/coreybutler/nvm-windows/issues/18)

**Reply by vtns-comment-importer** [reply]
```header
Originally posted by Bob on 2017-06-06 10:32
```

Yes Windows. I've found it's just simpler to open an elevated console :)

Just started dabbling in the SharePoint Framework - looking forward to your Voitanos Mastering SharePoint course being ready soon.

**Reply by vtns-comment-importer** [reply]
```header
Originally posted by Andrew Connell on 2017-06-06 10:42
```

Me too… “working on it…” ☺

**Comment by vtns-comment-importer** [comment]
```header
Originally posted by Prakash Thirunavukarasu on 2019-05-01 05:42
```

Hi Andrew,

I am strating gulp server and I am getting the below error in Windows server

D:\\SPFx\\FH-webpart>gulp serve  
Build target: DEBUG  
\[05:36:10\] Using gulpfile D:\\SPFx\\FH-webpart\\gulpfile.js  
\[05:36:10\] Starting gulp  
\[05:36:10\] Starting 'serve'...  
\[05:36:10\] Starting subtask 'configure-sp-build-rig'...  
\[05:36:10\] Finished subtask 'configure-sp-build-rig' after 6.79 ms  
\[05:36:10\] Starting subtask 'spfx-serve'...  
\[05:36:11\] Starting server...  
Starting api server on port 5432.  
Registring api: /workbench  
Registring api: \*/\*  
\[05:36:11\] Finished subtask 'spfx-serve' after 670 ms  
\[05:36:11\] Starting subtask 'pre-copy'...  
\[05:36:11\] Finished subtask 'pre-copy' after 60 ms  
\[05:36:11\] Starting subtask 'copy-static-assets'...  
\[05:36:11\] Starting subtask 'sass'...  
{ Error: listen EACCES: permission denied 127.0.0.1:4321  
at Server.setupListenHandle \[as \_listen2\] (net.js:1253:19)  
at listenInCluster (net.js:1318:12)  
at GetAddrInfoReqWrap.doListen \[as callback\] (net.js:1451:7)  
at GetAddrInfoReqWrap.onlookup \[as oncomplete\] (dns.js:61:10)  
code: 'EACCES',  
errno: 'EACCES',  
syscall: 'listen',  
address: '127.0.0.1',  
port: 4321 }  
\[05:36:11\] Error - Unknown  
listen EACCES: permission denied 127.0.0.1:4321  
\[05:36:11\] Opening https://localhost:5432/workbench using the default OS app

Error - Unknown  
listen EACCES: permission denied 127.0.0.1:4321

\[05:36:14\] Project fh-webpart version:0.0.1  
\[05:36:14\] Build tools version:3.9.7  
\[05:36:14\] Node version:v10.15.3  
\[05:36:14\] Total duration:7.87 s  
\[05:36:14\] Task errors:2  
\[05:36:14\] Error - 'undefined' errored after  
"orchestration aborted"

**Reply by vtns-comment-importer** [reply]
```header
Originally posted by Andrew Connell on 2019-05-01 13:48
```

Node.js v10 is not currently supported by SPFx.

**Comment by Adeybaby** [comment]
NVM has been my saviour in my failure to understand the world of Docker. I know they are not directly like for like, but the biggest part of keeping environments consistent for solutions is always (for me) Node. The NVM learning arc is tiny compared to the hours of Docker tutorials I would have sat through!

**Reply by andrewconnell** [reply]
I'm aware of that Docker option for doing SPFx dev, and I'm good friends with the guy behind it, but I think it's a terrible option for managing different environments. It adds way more complexity that the driver of their environment must be aware of to make it work... and there are always edge cases to deal with.

NVM just works, and it's elegantly simple. In fact, it does the exact same thing Node.js does when it runs.

- Install Node.js via an installer from their website, and it adds Node.js to your main app folder (Windows/MacOS), and updates your `PATH` environment variable to point to that location.
- ... or, use NVM to have multiple Node installs in different folders and it handles changing your `PATH` to switch versions.
